test/testenv/testenv_test.go

```diff
@@ -0,0 +1,127 @@
+package testenv
+
+import (
+	"flag"
+	"io"
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+	mixerapi "istio.io/api/mixer/v1"
+	"istio.io/mixer/adapter/denier"
+	"istio.io/mixer/pkg/attribute"
+	"istio.io/mixer/pkg/handler"
+)
+
+func closeHelper(c io.Closer) {
+	err := c.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func copyFile(dest, src string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	defer closeHelper(in)
+	out, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer closeHelper(out)
+	_, err = io.Copy(out, in)
+	return err
+
+}
+
+func buildTestConfigStore(relativePaths []string) (string, error) {
+	currentPath, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	configPath := path.Join(currentPath, uuid.New())
+	if err = os.Mkdir(configPath, os.ModePerm); err != nil {
+		return "", err
+	}
+
+	for _, filePath := range relativePaths {
+		err = copyFile(path.Join(configPath, path.Base(filePath)), path.Join(currentPath, filePath))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return configPath, nil
+}
+
+func getAttrBag(attribs map[string]interface{}) mixerapi.Attributes {
+	requestBag := attribute.GetMutableBag(nil)
+	requestBag.Set("target.service", "svc.cluster.local")
+	for k, v := range attribs {
+		requestBag.Set(k, v)
+	}
+
+	var attrs mixerapi.Attributes
+	requestBag.ToProto(&attrs, nil, 0)
+	return attrs
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	code := m.Run()
+	os.Exit(code)
+}
+
+func TestDenierAdapter(t *testing.T) {
+	configStore, err := buildTestConfigStore([]string{
+		"../../testdata/config/attributes.yaml",
```

Comment:
This is tricky.. I would recomment asking for text, creating file on the fly , putting them in temp dir and running from it.. Ideally adapter developer should just be writing TestDenierAdapter. For reference https://github.com/istio/mixer/pull/1176/files#diff-8349006db2c242fd7424e1dfb3295840R32

Reply:
This just serves as an example, so I use a checked-in config instead of have the config inline in test. But the test framework itself doesn't prevent adapter developer from doing what you describe. Once mem based config store is in, that should be preferred way.

Reply:
SG. Let's use the checked in config for this test. Later as we find more common patterns, we might find code that can belong to the test env framework

Reply:
This is fine for now, but when we go back to refactor we should pull these files in via bazel rather than relying on directory structure. (I know you pull in a file group, but there's some magic we can use to move the files to the same dir as the test at build/execution time.)

Longer term I'd like to see us move to an interface where the entire test body looks something like:
```go
client, _ := testenv.New(MyAdapter.GetInfo, "myfirstconfig.yaml", "mysecondconfig.yaml", ...)

attr := map[string]interface{}{"request.headers": map[string]string{"clnt": "abc"}}
bag := getAttrBag(attr)
request := mixerapi.CheckRequest{Attributes: bag}
resq, err := client.Check(context.Background(), &request)
if err != nil {
	t.Errorf("fail to send check to Mixer %v", err)
}
if resq.Precondition.Status.Code != 7 {
	t.Error(`resq.Precondition.Status.Code != 7`)
}
```

And all the cleanup is handled by the framework.

Reply:
Ideally, test framework should use mem based store so no clean up on FS is needed. But this would be post v0.2